js/index.js

```diff
@@ -0,0 +1,119 @@
+let books = JSON.parse(localStorage.getItem('books'));
+
+//  Error Message
+function errorMsg(message) {
+  document.querySelector('.warning').textContent = message;
+  setTimeout(() => {
+    document.querySelector('.warning').textContent = '';
+  }, 2000);
+}
+class Book {
+  constructor(title, author, id) {
+    this.title = title;
+    this.author = author;
+    this.id = id;
+  }
+
+  addBook() {
+    const { title, author, id } = this;
+    const bookList = { title, author, id };
+    books = JSON.parse(localStorage.getItem('books'));
+    if (title === '' || author === '') {
+      errorMsg('All fields are required');
+    } else if (books !== null) {
+      books.push(bookList);
+      localStorage.setItem('books', JSON.stringify(books));
+      books = JSON.parse(localStorage.getItem('books'));
+    } else {
+      books = [];
+      books.push(bookList);
```

Comment:
- [ OPTIONAL ]  Good work implementing the line that adds a book to an array. It is always better to use methods that don't mutate the array. Please always prefer a method that does not mutate the array whenever possible. for a full list of array methods, please check this[ link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) 🤝